plugins/source/azure/resources/services/security/settings.go

```diff
@@ -1,86 +1,93 @@
+// Auto generated code - DO NOT EDIT.
+
 package security
 
 import (
 	"context"
 	"fmt"
 
 	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
-	"github.com/cloudquery/cq-provider-sdk/provider/diag"
 	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
 )
 
-func SecuritySettings() *schema.Table {
+func Settings() *schema.Table {
 	return &schema.Table{
-		Name:         "azure_security_settings",
-		Description:  "Setting the kind of the security setting",
-		Resolver:     fetchSecuritySettings,
-		Multiplex:    client.SubscriptionMultiplex,
-		DeleteFilter: client.DeleteSubscriptionFilter,
-		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"subscription_id", "id"}},
+		Name:      "azure_security_settings",
+		Resolver:  fetchSecuritySettings,
+		Multiplex: client.SubscriptionMultiplex,
 		Columns: []schema.Column{
 			{
-				Name:        "subscription_id",
-				Description: "Azure subscription id",
-				Type:        schema.TypeString,
-				Resolver:    client.ResolveAzureSubscription,
+				Name:     "subscription_id",
+				Type:     schema.TypeString,
+				Resolver: client.ResolveAzureSubscription,
 			},
 			{
-				Name:        "kind",
-				Description: "Possible values include: 'KindSetting', 'KindDataExportSettings'",
-				Type:        schema.TypeString,
+				Name:     "kind",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Kind"),
 			},
 			{
-				Name:        "id",
-				Description: "Resource Id",
-				Type:        schema.TypeString,
-				Resolver:    schema.PathResolver("ID"),
+				Name:     "id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ID"),
+				CreationOptions: schema.ColumnCreationOptions{
+					PrimaryKey: true,
+				},
 			},
 			{
-				Name:        "name",
-				Description: "Resource name",
-				Type:        schema.TypeString,
+				Name:     "name",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Name"),
 			},
 			{
-				Name:        "resource_type",
-				Description: "Resource type",
-				Type:        schema.TypeString,
-				Resolver:    schema.PathResolver("Type"),
+				Name:     "type",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Type"),
 			},
 			{
-				Name:        "enabled",
-				Description: "Export setting enabled flag",
-				Type:        schema.TypeBool,
-				Resolver:    schema.PathResolver("DataExportSettingProperties.Enabled"),
```

Comment:
There's a bug here. We can't statically resolve the enable field as `Settings` has different sub structs `DataExportSetting` and `AlertSyncSettings`, see `resolveEnabled`